server/services/EventService.js

```diff
@@ -0,0 +1,38 @@
+const Event = require('../models/Event');
+const WalletService = require('./WalletService');
+const Session = require('../infra/database/Session');
+const HttpError = require('../utils/HttpError');
+
+class EventService {
+  constructor() {
+    this._session = new Session();
+    this._event = new Event(this._session);
+    this._walletService = new WalletService();
+  }
+
+  async getAllEvents({ wallet, limit, since, walletLoginId }) {
+    let events = [];
+
+    if (wallet) {
+      const walletInstance = await this._walletService.getByName(wallet);
+      const isSub = await this._walletService.hasControlOver(
+        walletLoginId,
+        walletInstance.id,
+      );
+      if (!isSub) {
+        throw new HttpError(
+          403,
+          'Wallet does not belong to the logged in wallet',
+        );
+      }
```

Comment:
Can we have this as a reusable function?

Reply:
I added this into the walletService as hasControlOverByName